lib/Canonicalize/MakeModelGEPPass.cpp

```diff
@@ -1219,10 +1196,11 @@ computeBestInArray(const model::QualifiedType &BaseType,
   auto ArrayAccessInfo = ChildInfo{ .Index = std::move(ElementSummation),
                                     .Type = AggregateKind::Array };
 
-  ModelGEPReplacementInfo BestInArray(BaseArray,
-                                      { ArrayAccessInfo },
-                                      SummationInElement,
-                                      ElementType);
+  ModelGEPReplacementInfo
+    BestInArray(/* Base = */ BaseType.copy(),
```

Comment:
I've to say I'm not a huge fan of `X.copy()` having the exact same effect as `X`.

It's like doing...
```c++
void myfunc(int);

int main() {
  int Asd = 3;
  myfunc(Asd.copy());
}
```

For instance, below we have:
```c++
  auto ElementResult = rc_recur computeBest(Array.ElementType(),
                                            SummationInElement,
                                            AccessedTypeOnIR,
                                            VH);
```
Could we have `Array.ElementType().copy()` there too? It's confusing.

Do we really really really want it?

Reply:
But it's not the same since the constructor is
```cpp
  ModelGEPReplacementInfo(model::UpcastableType &&Base,
                          const ChildIndexVector &Indices,
                          const IRSummation &Mismatch,
                          model::UpcastableType &&Accessed)
```

> Do we really really really want it?

Once we update to the next clang version, we can drop it since we'll have the standard way to express it: `BestInArray(/* Base = */ auto(BaseType),`

Reply:
As per our discussion, let's move to value semantics (i.e., copy first).

Reply:
Done.